src/govuk/components/error-summary/error-summary.mjs

```diff
@@ -99,16 +99,20 @@ ErrorSummary.prototype.handleClick = function (event) {
  * NVDA (as tested in 2018.3.2) - without this only the field type is announced
  * (e.g. "Edit, has autocomplete").
  *
- * @param {HTMLElement} $target - Event target
+ * @param {HTMLElement | HTMLAnchorElement} $target - Event target
```

Comment:
Yeah you can do an inline `/** @type {HTMLAnchorElement} */`

But that only removes the compiler warning, won't do anything to stop your code crashing.
 
To be "type correct" we'd actually want to go with:
 
```suggestion
 * @param {EventTarget} $target - Event target
```

But then if we were confident enough in IE8 support you'd guard that with:

```mjs
if ($target instanceof HTMLAnchorElement) {
```

I picked `{HTMLElement | HTMLAnchorElement}` as it shows the route the code takes—filtering out "non-link" HTML elements so we're only left with `HTMLAnchorElement `, but I'm open to adjusting things like this further?

Can move these to another PR too

Reply:
> But that only removes the compiler warning, won't do anything to stop your code crashing.

I think there's the first early return that ensures we don't blow up if we're given something that's not an HTMLAnchorElement (using duck-typing)

> I picked {HTMLElement | HTMLAnchorElement} as it shows the route the code takes—filtering out "non-link" HTML elements so we're only left with HTMLAnchorElement

Is that a convention when using types, for when the listed types extend each other?

Reply:
@romaricpascal Yeah this one's fine (won't crash), I mean as a general rule elsewhere.

Don't cast unless you have to

Not really a convention no

* If we're being strict, it's wrong (should be `EventTarget`)
* If we're being loose, it does document the expected types better

It's annoying because [`EventTarget`](https://dom.spec.whatwg.org/#eventtarget) is one of the lower level interfaces, 

```ts
HTMLAnchorElement extends HTMLElement
HTMLElement extends Element
Element extends Node
Node extends EventTarget
```

So without using `instanceof` the compiler gives up, it has no idea what extends `EventTarget`

<img width="594" alt="Example showing EventTarget type guarded by tagName check" src="https://user-images.githubusercontent.com/415517/201084256-d447b13d-dc61-4a72-87e8-19d56ae350bc.png">

But if we had IE8 compatibility for `instanceof HTMLAnchorElement` the compiler can continue, spotting other bugs/issues and compatibility problems in your code 🙌 

<img width="874" alt="Example showing EventTarget type guarded by instanceof" src="https://user-images.githubusercontent.com/415517/201084523-ca9255cb-0b11-4768-a341-988a4018895e.png">

So I picked `{HTMLElement | HTMLAnchorElement}` as the middle ground.

Now I've talked it through I think I'll change it back to `{EventTarget}` 😆 

Let's save all these improvements for another day

Reply:
Had another play, seems like `{EventTarget | HTMLAnchorElement}` is the best compromise

Reply:
Done + Moved to PR:

* https://github.com/alphagov/govuk-frontend/pull/2997

Reply:
Additional note to say we've moved higher up the `instanceof` inheritance chain and stuck with `Element`